bears/js/ESLintBear.py

```diff
@@ -0,0 +1,77 @@
+import json
+
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Diff import Diff
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+from coalib.misc.Shell import escape_path_argument
+from coalib.results.Result import Result
+from json.decoder import JSONDecodeError
+
+
+class ESLintBear(LocalBear, Lint):
+    executable = 'eslint'
+    severity_map = {
+        "2": RESULT_SEVERITY.MAJOR,
+        "1": RESULT_SEVERITY.NORMAL,
+        "0": RESULT_SEVERITY.INFO
+    }
+    use_stdin = True
+
+    def run(self, filename, file, eslint_config: str=""):
+        '''
+        Checks the code with eslint. This will run eslint over each of the files
+        seperately.
+
+        :param eslint_config: The location of the .eslintrc config file.
+        '''
+        self.arguments = '--no-ignore --no-color -f=json --stdin'
+        if eslint_config:
+            self.arguments += (" --config "
+                               + escape_path_argument(eslint_config))
+
+        return self.lint(filename, file)
+
+    def _process_issues(self, output, filename):
+        try:
+            output = json.loads("".join(output))
+            lines = [line for line in open(filename)]
+            lines = "".join(lines)
+            newoutput = lines
+            replacement_count = 0
+            for lint_result in output[0]['messages']:
+                if lint_result['severity'] == 0:
+                    continue
+                if 'fix' in lint_result:
+                    fixStructure = lint_result['fix']
+                    startingValue, endingValue = fixStructure['range']
+                    replacementText = fixStructure['text']
+                    if replacementText != "":
+                        replacement_count+=1
+                    newoutput = newoutput[
+                        :int(startingValue)+replacement_count] + replacementText +\
+                        newoutput[int(endingValue+replacement_count):]
+                    # print(newoutput)
+                else:
+                    yield Result.from_values(
+                        origin=self,
+                        message=lint_result['message'],
+                        file=filename,
+                        severity=self.severity_map[str(lint_result['severity'])],
+                        line=lint_result['line'])
+                    pass
+
+            diff = Diff.from_string_arrays(
+                lines.splitlines(True), newoutput.splitlines(True))
+
+            yield Result.from_values(
+                origin=self,
+                message=lint_result['message'],
+                file=filename,
+                diffs={filename: diff},
+                severity=self.severity_map[str(lint_result['severity'])],
+                line=lint_result['line'])
+        except JSONDecodeError:
```

Comment:
The JSONDecodeError is only raised in `json.loads` so why add everything into try ?